### PR TITLE
issue/2553 Added a11y_alt_text helper

### PR DESCRIPTION
**Pull request by oliverfoster**
#2553 
* Added helper to allow alternative text to read instead

This helper
```html
{{a11y_alt_text '$5bn' 'five billion dollars'}}
{{a11y_alt_text '$5bn' 'five' 'billion' 'dollars'}}
```
Produces
```html
<span aria-hidden="true">$5bn</span><span class="aria-label">five billion dollars</span>
```

### CODE REVIEW

File: src/core/js/helpers.js [hunk 1]
```diff
@@ -355,6 +355,17 @@ define([
         a11y_attrs_tabbable: function() {
             Adapt.a11y.log.deprecated('a11y_attrs_tabbable should not be used. tabbable elements should be natively tabbable.');
             return new Handlebars.SafeString(' role="region" tabindex="0" ');
+        },
+
+        /**
+         * Produce display text with alternative screen reader version.
+         * @param {string} visible Visual text
+         * @param {string} alternatives Any number of additional arguments which will be concatenated and read
```
Comment (moloko) [inline src/core/js/helpers.js]: an `@usage` example would be helpful here e.g. `@usage {{a11y_alt_text '$5bn' 'five billion dollars'}}`

File: src/core/js/helpers.js [hunk 2]
```diff
@@ -355,6 +355,18 @@ define([
         a11y_attrs_tabbable: function() {
             Adapt.a11y.log.deprecated('a11y_attrs_tabbable should not be used. tabbable elements should be natively tabbable.');
             return new Handlebars.SafeString(' role="region" tabindex="0" ');
+        },
+
+        /**
+         * Produce display text with alternative screen reader version.
+         * @param {string} visible Visual text
```
Comment (moloko) [inline src/core/js/helpers.js]:
```suggestion
         * @param {string} visible Text that will be displayed on screen
```

File: src/core/js/helpers.js [hunk 3]
```diff
@@ -355,6 +355,18 @@ define([
...
+         * @param {string} alternatives Any number of additional arguments which will be concatenated and read
```
Comment (moloko) [inline src/core/js/helpers.js]: Text that will be read by the screen reader (instead of what's displayed on screen)

**Comment by oliverfoster** [comment]
```html
{{a11y_alt_text 'Here are some bits to read', 'There are', _items.length, 'items to read'}}
```

**Comment by moloko** [comment]
> ```
> {{a11y_alt_text 'Here are some bits to read' 'There are' _items.length 'items to read'}}
> ```
> 
> It gives a bit more flexibility for mixing in variables for complex text and fits with some of the other helpers.

Ah yes I get it now! Of course you know what I'm going to ask you do next? Yep, add that as another usage example ;-)

File: src/core/js/helpers.js [hunk 4]
```diff
@@ -362,6 +362,7 @@ define([
          * @param {string} visible Text that will be displayed on screen
          * @param {string} alternatives Text that will be read by the screen reader (instead of what's displayed on screen)
          * @usage {{a11y_alt_text '$5bn' 'five billion dollars'}}
+         * @usage {{a11y_alt_text '$5bn' 'five' 'billion' 'dollars'}}
```
Comment (moloko) [inline src/core/js/helpers.js]: sorry but I meant the more recent example, also you don't have to put two @usage entries in you can just do it like this:
> @usage `{{a11y_alt_text '$5bn' 'five billion dollars'}}` or `{{a11y_alt_text 'Here are some bits to read' 'There are' _items.length 'items to read'}}`

File: src/core/js/helpers.js [hunk 5]
```diff
@@ -355,6 +355,18 @@ define([
         a11y_attrs_tabbable: function() {
             Adapt.a11y.log.deprecated('a11y_attrs_tabbable should not be used. tabbable elements should be natively tabbable.');
             return new Handlebars.SafeString(' role="region" tabindex="0" ');
+        },
+
+        /**
+         * Produce display text with alternative screen reader version.
+         * @param {string} visible Text that will be displayed on screen
+         * @param {string} alternatives Text that will be read by the screen reader (instead of what's displayed on screen)
+         * @usage {{a11y_alt_text '$5bn' 'five billion dollars'}} or {{a11y_alt_text 'Here are some bits to read' 'There are' _items.length 'items to read'}}
```
Comment (moloko) [inline src/core/js/helpers.js]: sorry totally my mistake
```suggestion
         * @example {{a11y_alt_text '$5bn' 'five billion dollars'}} or {{a11y_alt_text 'Here are some bits to read' 'There are' _items.length 'items to read'}}
```

**Comment by oliverfoster** [comment]
@moloko this should be ready now hopefully?

**Pull request merged**